connector-sdk/document/src/main/java/io/camunda/operation/DefaultIntrinsicOperationExecutor.java

```diff
@@ -0,0 +1,113 @@
+  private final ObjectMapper objectMapper;
+
+  public DefaultIntrinsicOperationExecutor(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+    this.parameterBinder = new IntrinsicOperationParameterBinder(objectMapper);
+
+    final var operationProviders =
+        ServiceLoader.load(IntrinsicOperationProvider.class).stream()
+            .map(ServiceLoader.Provider::get)
+            .toList();
+
+    if (operationProviders.isEmpty()) {
+      LOGGER.warn(
+          "No intrinsic operation providers found. "
+              + "Please make sure to provide at least one implementation of IntrinsicOperationProvider.");
+      return;
+    }
+
+    for (var provider : operationProviders) {
+      final var methodsByOperationName = getDeclaredMethods(provider);
+      for (var entry : methodsByOperationName.entrySet()) {
+        if (operationSources.containsKey(entry.getKey())) {
+          throw new IllegalArgumentException(
+              "Operation with name: "
+                  + entry.getKey()
+                  + " duplicated in providers: "
+                  + provider.getClass().getName()
+                  + " and "
+                  + operationSources.get(entry.getKey()).provider.getClass().getName());
+        }
+        operationSources.put(entry.getKey(), new OperationSource(provider, entry.getValue()));
+      }
+    }
+  }
+
+  @Override
+  public <T> T execute(String operationName, IntrinsicOperationParams params, Class<T> resultType) {
+    final var source = operationSources.get(operationName);
+    if (source == null) {
+      throw new IllegalArgumentException("No operation found with name: " + operationName);
+    }
+    final var arguments = parameterBinder.bindParameters(source.method, params);
+    try {
+      final Object result = source.method.invoke(source.provider, arguments);
+      return objectMapper.convertValue(result, resultType);
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Failed to execute operation: "
+              + operationName
+              + " and retrieve a result of type: "
+              + resultType.getName(),
+          e);
+    }
+  }
+
+  private Map<String, Method> getDeclaredMethods(IntrinsicOperationProvider provider) {
+    final var methodsByOperationName =
+        Arrays.stream(provider.getClass().getDeclaredMethods())
+            .filter(method -> method.isAnnotationPresent(IntrinsicOperation.class))
+            .collect(
+                Collectors.toMap(
+                    method -> method.getAnnotation(IntrinsicOperation.class).name(),
+                    method -> method));
+    if (methodsByOperationName.isEmpty()) {
+      throw new IllegalArgumentException(
+          "No intrinsic operations found in provider "
+              + provider.getClass().getName()
+              + ". "
+              + "At least one method with @IntrinsicOperation expected.");
+    }
+    return methodsByOperationName;
+  }
```

Comment:
We can refactor this later but I would extract the discovery and reflection part from the executor.

A `IntrinsicOperationRegistry` could do the discovery and do the reflection resulting in a dedicated `IntrinsicOperationInstance` record that hold all the necessary information for the executor to just "call" it.

Reply:
Done